flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPool.java

```diff
@@ -824,11 +814,8 @@ private void failPendingRequest(PendingRequest pendingRequest, Exception e) {
 	}
 
 	@Nullable
-	private SlotAndLocality pollAndAllocateSlot(
-		SlotRequestId slotRequestId,
-		ResourceProfile resourceProfile,
-		Collection<TaskManagerLocation> locationPreferences) {
-		SlotAndLocality slotFromPool = availableSlots.poll(resourceProfile, locationPreferences);
+	private SlotAndLocality pollAndAllocateSlot(SlotRequestId slotRequestId, SlotProfile slotProfile) {
```

Comment:
Would it be enough to pass in the `ProfileToSlotContextMatcher` instead of the `SlotProfile`?

Reply:
Then we would miss the resource profile, right?